lib/vagrant-libvirt/templates/domain.xml.erb

```diff
@@ -152,11 +152,13 @@
 <%- @domain_volumes.each_with_index do |volume, index| -%>
     <disk type='file' device='disk'>
       <alias name='ua-box-volume-<%= index -%>'/>
-      <driver name='qemu' type='qcow2' <%=
-        @disk_driver_opts.empty? ? "cache='#{volume[:cache]}'" :
-            @disk_driver_opts.reject { |k,v| v.nil? }
-                             .map { |k,v| "#{k}='#{v}'"}
-                             .join(' ') -%>/>
+      <driver name='qemu'<%=
+        '%{type}%{cache}%{io}%{copy_on_read}%{discard}%{detect_zeroes}' \
+          % { :type => 'qcow2', :cache => "#{volume[:cache]}", :io => nil, :copy_on_read => nil, :discard => nil, :detect_zeroes => nil }
+            .merge(@disk_driver_opts)
+            .select { |k,_| [:type, :cache, :io, :copy_on_read, :discard, :detect_zeroes].include? k }
```

Comment:
The valid options are already controlled from config.rb, I'd prefer not to be hardcoding them here again.

Reply:
Ah yes, I see how this is redundant.